scripts/SubsetSelection/src/encoders/qwen2.py

```
@@ -0,0 +1,313 @@
+            Dataset: The dataset with an added 'embedding' column.
+        """
+        texts = dataset[column_name]
+        embeddings = self.encode(inputs=texts, instruction=instruction, query_description=query_description)
+        
+        # Convert tensors to lists
+        embeddings = embeddings.cpu().numpy().tolist()
+        return dataset.add_column("embedding", embeddings)
+        
+if __name__ == "__main__":
+    # Test the encoder
+    encoder = Qwen2EmbedDecoder(
```

Ruff (0.13.1)

Undefined name `Qwen2EmbedDecoder`

(F821)